pallets/multisig/src/benchmarking.rs

```diff
@@ -0,0 +1,443 @@
+
+#![cfg(feature = "runtime-benchmarks")]
+use crate::*;
+use frame_benchmarking::benchmarks;
+use frame_system::RawOrigin;
+use polymesh_common_utilities::benchs::UserBuilder;
+
+pub type MultiSig<T> = crate::Module<T>;
+pub type Identity<T> = identity::Module<T>;
+pub type Timestamp<T> = pallet_timestamp::Module<T>;
+
+fn generate_signers<T: Trait>(signers: &mut Vec<Signatory<T::AccountId>>, n: usize) {
+    signers.extend((0..n).map(|x| {
+        Signatory::Account(
+            <UserBuilder<T>>::default()
+                .seed(x as u32)
+                .build("key")
+                .account,
+        )
+    }));
+}
+
+fn get_last_auth_id<T: Trait>(signatory: &Signatory<T::AccountId>) -> u64 {
+    <identity::Authorizations<T>>::iter_prefix_values(signatory)
+        .into_iter()
+        .max_by_key(|x| x.auth_id)
+        .expect("there are no authorizations")
+        .auth_id
+}
+
+fn generate_multisig<T: Trait>(
+    alice: T::AccountId,
+    origin: RawOrigin<T::AccountId>,
+    signers: Vec<Signatory<T::AccountId>>,
+) -> Result<T::AccountId, DispatchError> {
+    let num_of_signers = signers.len() as u64;
+    generate_multisig_with_signers::<T>(alice, origin, signers, num_of_signers)
+}
+
+fn generate_multisig_with_signers<T: Trait>(
+    alice: T::AccountId,
+    origin: RawOrigin<T::AccountId>,
+    signers: Vec<Signatory<T::AccountId>>,
+    num_of_signers: u64,
+) -> Result<T::AccountId, DispatchError> {
+    let multisig = <MultiSig<T>>::get_next_multisig_address(alice.clone());
+    <MultiSig<T>>::create_multisig(origin.into(), signers.clone(), num_of_signers)?;
+    for signer in signers {
+        let auth_id = get_last_auth_id::<T>(&signer);
+        <MultiSig<T>>::unsafe_accept_multisig_signer(signer, auth_id)?;
+    }
+    Ok(multisig)
+}
+
+const MAX_SIGNERS: u32 = 256;
+
+benchmarks! {
+    _ {}
+
+    create_multisig {
+        // Number of signers
+        let i in 1 .. MAX_SIGNERS;
+
+        let caller = <UserBuilder<T>>::default().generate_did().build("caller");
+        let mut signers = vec![Signatory::from(caller.did())];
+        generate_signers::<T>(&mut signers, i as usize);
+        let multisig = <MultiSig<T>>::get_next_multisig_address(caller.account());
+    }: _(caller.origin(), signers, i as u64)
+    verify {
+        ensure!(<MultiSigToIdentity<T>>::contains_key(multisig), "create_multisig");
+    }
+
+    create_or_approve_proposal_as_identity_create {
+        let alice = <UserBuilder<T>>::default().generate_did().build("alice");
+        let mut signers = vec![Signatory::from(alice.did())];
+        generate_signers::<T>(&mut signers, 1);
+        let multisig = generate_multisig::<T>(alice.account(), alice.origin(), signers)?;
+        let proposal_id = <MultiSig<T>>::ms_tx_done(multisig.clone());
+    }: create_or_approve_proposal_as_identity(alice.origin(), multisig.clone(), Box::new(frame_system::Call::<T>::remark(vec![]).into()), Some(<Timestamp<T>>::get() + 1.into()), true)
```

Comment:
Let's extract the proposal creation into a function and reuse it in all cases.
Also, let's extract the expiry creation. This could be added into the function above, returning a tuple.
Finally, this should be done as part of setup, and not in the `extrinsic(args)` part of the bench, as anything done in `args` actually contributes to the weight, which is wrong.